Zombies? zombies!

kiVts opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

# About the pull request
_Oh god its kivts again_
okay so, zombies, lots and lots and lots of things, still wip because Im missing sprites for actions(currently using headbite sprite for both), but it works(UPD, the action buttons not updating, will fix later huh).

what it does
- gives zombies two xeno like actions, bite and leap. leap have 10 second cooldown, probabbly not the only ones in the future, but currently - yes.
- bite infects with 100% chance, 3 second windup, 6 second cooldown
- leap is a slow runner pounce (literally pounce code)
- nerfed zombies bullet resistance and their claws(40>30)
- spaceacillin given a purpose, it can(key word CAN, its just 3 percent chance to heal on level one) now heal stage one zombie virus, also while at it hardens the recipe by a quite a bit and makes the od much much more restrictive
- antibiotic property, same as above but can be created by research(and be made better!).
- removes ""what abouts" information about zombie virus for dead mobs
- removed around 10 if(prob) in black_goo code and replaced it with pick() so its hopefully better
- expands on human abilities a bit to be better for zombies
- removed slash on infect entirely, only bites can infect
- Fixed  1 hour long revive timer

reviving timer?  ~~havent touched that yet~~ (this shit is fixed now, if correct)

<!-- Remove this text and explain what the purpose of your PR is.

Mention if you have tested your changes. If you changed a map, make sure you used the mapmerge tool.
If this is an Issue Correction, you can type "Fixes Issue #169420" to link the PR to the corresponding Issue number #169420.

Remember: something that is self-evident to you might not be to others. Explain your rationale fully, even if you feel it goes without saying. -->

# Explain why it's good for the game

someone wanted to remove it so here we are

# Testing Photographs and Procedure
<summary>Screenshots & Videos</summary>

I tried to record gifs but that went horribly for god knows what reason , so **NO GIFS FOR YOU**
</details>


# Changelog

:cl: kivts
add: Two new xeno-like abilities to zombies
add: spaceacillin has a small chance to cure zombie virus
code: Black goo code is now generally better
/:cl:

<!-- Both :cl:'s are required for the changelog to work! -->

carlarctg commented:
holy shit

Agameofscones reviewed (approved):
sprite approved

kiVts commented:
hold on but its reusing the same ico- ah well

kiVts commented:
gonna go into draft because its certainly not over

kiVts commented:
-removed zombie chance to infect on hit entirely, you can now only infect with bites
-fixed the thing with action buttons update, and it now uses pounce iccon
-spaceacillin now slows down progression of the virus at stage two and stage three. on stage two it halfs the progression rate and on stage three its * 0.75. stage 4 untouched
- adjustments to disease.dm to add the above

kiVts commented:
on the matter of revive timers being very long, I have no idea whats causing this since on local it works perfectly even with 20 zombs going revival process at the same time. probably something with server!!

kiVts commented:
you can remotly bite people as zombie. will fix this later

kiVts commented:
okay, fixed some cooldown jokes I had, also zombies cant harm dead mobs(still can bite though!), also added cooldown messages like in xeno actions. I think Its ready

Diegoflores31 commented:
> okay, fixed some cooldown jokes I had, also zombies cant harm dead mobs(still can bite though!), also added cooldown messages like in xeno actions. I think if nothing comes up its ready

what if abilities had a cost ? like you get powerfull the more people you murder ex. extra range on your leap or less cooldown and maybe bite could be something that is like 10 seconds and goes up to 5 depending on your kills

kiVts commented:
> > okay, fixed some cooldown jokes I had, also zombies cant harm dead mobs(still can bite though!), also added cooldown messages like in xeno actions. I think if nothing comes up its ready
> 
> what if abilities had a cost ? like you get powerfull the more people you murder ex. extra range on your leap or less cooldown and maybe bite could be something that is like 10 seconds and goes up to 5 depending on your kills

so like abom? That sounds easy.

harryob commented on code/datums/disease.dm:
```diff
@@ -42,10 +42,10 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
 	var/stage_minimum_age = 0 // how old the disease must be to advance per stage
 	var/survive_mob_death = FALSE //whether the virus continues processing as normal when the affected mob is dead.
-
+	var/age_multiplier = 1 //Helpfull to slow down progression of virruses, <1 is slower, >1 is faster. 1 is default
```
```suggestion
	/// Helpful to slow down progression of viruses, <1 is slower, >1 is faster. 1 is default
	var/age_multiplier = 1 
```

harryob commented on code/datums/diseases/black_goo.dm:
```diff
@@ -23,66 +23,63 @@
-				if(prob(3))
-					to_chat(affected_mob, SPAN_DANGER("You feel really warm..."))
-					goo_message_cooldown = world.time + 100
+				var/message = pick("Your mouth feels really dry..", "Your head aches...",)
```
```suggestion
				var/message = pick("Your mouth feels really dry...", "Your head aches...",)
```

harryob commented on code/datums/diseases/black_goo.dm:
```diff
@@ -23,66 +23,63 @@
+			H.nutrition = NUTRITION_VERYLOW //brains tasty yey :D
 			H.next_move_slowdown = max(H.next_move_slowdown, 2)
-			if(prob(5) || age >= stage_minimum_age-1)
+			var/message = pick("MAKE IT STOP", "Your skin is pulling itself apart!", "ITS OVER SOON")
```
```suggestion
			var/message = pick("MAKE IT STOP!", "Your skin is pulling itself apart!", "IT'S OVER SOON!")
```

harryob commented on code/datums/diseases/black_goo.dm:
```diff
@@ -131,26 +137,19 @@
 		return FALSE
-
+	if(target.stat == DEAD)
+		to_chat(user, SPAN_XENOWARNING("[target.name] is already dead, why would you want to do that?"))
```
```suggestion
		to_chat(user, SPAN_XENOWARNING("[target] is already dead, why would you want to do that?"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+	if(get_dist(zombie, M) > 1)
+		return
+	var/mob/living/carbon/human/victim = M
+	var/datum/disease/black_goo/D = locate() in victim.viruses
+	if(D)
```
D -> located_disease

kiVts replied:
done

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+	var/mob/living/carbon/human/victim = M
+	var/datum/disease/black_goo/D = locate() in victim.viruses
+	if(D)
+		to_chat(zombie, SPAN_XENOWARNING("the target is already infected, why would you bite it again?"))
```
```suggestion
		to_chat(zombie, SPAN_XENOWARNING("The target is already infected, why would you bite it again?"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+		enter_cooldown(10 SECONDS)
+		to_chat(zombie, SPAN_XENOWARNING("You reach down to [M.name] neck, preparing to bite it!"))
+		to_chat(M, SPAN_DANGER("[zombie.name] reaches to your neck and starts to open his jaw, agh!"))
+		if(do_after(zombie, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, M, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
```
early return

ie
```dm
if(!do_after)
 to_chat(zombie, "interrupted")
  return

to_chat(zombie, SPAN_XENOWARNING...
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+			zomb.anchored = FALSE
+			zomb.update_canmove()
+
+	zomb.visible_message(SPAN_XENOWARNING("\The [zomb.name] [name][findtext(name, "e", -1) || findtext(name, "p", -1) ? "s" : "es"] at [target]!"), SPAN_XENOWARNING("You [name] at [target]!"))
```
```suggestion
	zomb.visible_message(SPAN_XENOWARNING("[zomb] [name][findtext(name, "e", -1) || findtext(name, "p", -1) ? "s" : "es"] at [target]!"), SPAN_XENOWARNING("You [name] at [target]!"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+		if(do_after(zombie, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, M, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
+			to_chat(zombie, SPAN_XENOWARNING("You bite [M.name] neck, leaving a bloody mark!"))
+			to_chat(M, SPAN_DANGER("[zombie.name] Bites you right in the neck!"))
+			playsound(zombie.loc, 'sound/hallucinations/wail.ogg', 25, TRUE)
```
```suggestion
			playsound(get_turf(zombie), 'sound/hallucinations/wail.ogg', 25, TRUE)
```
why not just play sound from the zombie though

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+		to_chat(M, SPAN_DANGER("[zombie.name] reaches to your neck and starts to open his jaw, agh!"))
+		if(do_after(zombie, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, M, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
+			to_chat(zombie, SPAN_XENOWARNING("You bite [M.name] neck, leaving a bloody mark!"))
+			to_chat(M, SPAN_DANGER("[zombie.name] Bites you right in the neck!"))
```
```suggestion
			to_chat(M, SPAN_DANGER("[zombie] bites you right in the neck!"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+		to_chat(zombie, SPAN_XENOWARNING("You reach down to [M.name] neck, preparing to bite it!"))
+		to_chat(M, SPAN_DANGER("[zombie.name] reaches to your neck and starts to open his jaw, agh!"))
+		if(do_after(zombie, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, M, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
+			to_chat(zombie, SPAN_XENOWARNING("You bite [M.name] neck, leaving a bloody mark!"))
```
```suggestion
			to_chat(zombie, SPAN_XENOWARNING("You bite [M] neck, leaving a bloody mark!"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+	else
+		enter_cooldown(10 SECONDS)
+		to_chat(zombie, SPAN_XENOWARNING("You reach down to [M.name] neck, preparing to bite it!"))
+		to_chat(M, SPAN_DANGER("[zombie.name] reaches to your neck and starts to open his jaw, agh!"))
```
```suggestion
		to_chat(M, SPAN_DANGER("[zombie.name] reaches your neck and starts to open [zombie.p_their()] jaw!"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+		to_chat(zombie, SPAN_XENOWARNING("the target is already infected, why would you bite it again?"))
+	else
+		enter_cooldown(10 SECONDS)
+		to_chat(zombie, SPAN_XENOWARNING("You reach down to [M.name] neck, preparing to bite it!"))
```
```suggestion
		to_chat(zombie, SPAN_XENOWARNING("You reach down to [M] neck, preparing to bite it!"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -170,7 +172,17 @@ CULT
+	if(!owner)
+		return
+	update_button_icon()
+	if(cooldown_message_on)
+		if(isnull(cooldown_message))
+			to_chat(owner, SPAN_XENODANGER("You feel your strength return! You can use [name] again!"))
+		else
```
i don't quite get why this is like this, if it's just copypasta let me know but uhhh... we have TIMER_COOLDOWN_START and stuff

kiVts replied:
I really just looked at xeno actions and looked at human_abilities... yeha

harryob commented on code/modules/mob/living/carbon/human/species/zombie.dm:
```diff
@@ -24,9 +24,12 @@
-
 	has_species_tab_items = TRUE
-
+	var/d
```
erm?

kiVts replied:
I have no idea either

harryob commented on code/modules/reagents/chemistry_properties/prop_positive.dm:
```diff
@@ -900,6 +900,35 @@
+	if(D)
+		var/datum/disease/black_goo/virus = new
+		var/mob/living/carbon/human/H = M
+		switch(virus.stage)
```
does this need to be a switch if there's only one condition

harryob commented on code/modules/reagents/chemistry_properties/prop_positive.dm:
```diff
@@ -900,6 +900,35 @@
+	var/datum/disease/black_goo/D = locate() in M.viruses
+	if(D)
+		var/datum/disease/black_goo/virus = new
+		var/mob/living/carbon/human/H = M
```
no one letter vars...

harryob commented on code/modules/reagents/chemistry_properties/prop_positive.dm:
```diff
@@ -900,6 +900,35 @@
+
+/datum/chem_property/positive/antibiotic/process(mob/living/M, potency = 1, delta_time)
+	M.apply_damage(0.5 * potency * delta_time, TOX)
+	var/datum/disease/black_goo/D = locate() in M.viruses
```
no one letter vars...

harryob commented on code/datums/diseases/black_goo.dm:
```diff
@@ -23,66 +23,63 @@
+				age_multiplier = 1
+			hidden = list(1,0)
 			H.next_move_slowdown = max(H.next_move_slowdown, 1)
+			var/message = pick("Your muscles ache", "Your throat is really dry", "Skin on your hands peels away.", "You can feel your heart skipping a beat...")
```
```suggestion
			var/message = pick("Your muscles ache.", "Your throat is really dry.", "Skin on your hands peels away.", "You can feel your heart skipping a beat...")
```

harryob commented on code/datums/diseases/black_goo.dm:
```diff
@@ -94,10 +91,18 @@
 				H.nutrition = NUTRITION_MAX //never hungry
 
+/datum/disease/black_goo/proc/victim_has_antibiotic()
+	for(var/i=1 to affected_mob.reagents.reagent_list.len)
+		var/datum/reagent/S = affected_mob.reagents.reagent_list[i]
+		if(S.get_property(PROPERTY_ANTIBIOTIC))
+			return TRUE
+		else
+			return FALSE
```
```suggestion
	for(var/datum/reagent/current_reagent as anything in affected_mob.reagents.reagent_list)
		if(current_reagent.get_property(PROPERTY_ANTIBIOTIC))
			return TRUE
	return FALSE
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+		if(!do_after(zombie, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, M, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
+			to_chat(zombie, SPAN_XENOWARNING("You were interupted!"))
+			return
+		to_chat(zombie, SPAN_XENOWARNING("You bite [M] neck, leaving a bloody mark!"))
```
```suggestion
		to_chat(zombie, SPAN_XENOWARNING("You bite [M]'s neck, leaving a bloody mark!"))
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+
+
+/datum/action/human_action/activable/zombie/leap
+	name = "leap"
```
```suggestion
	name = "Leap"
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,109 @@ CULT
+/datum/action/human_action/activable/zombie/leap
+	name = "leap"
+	action_icon_state = "leap"
+	var/maxdistance = 5 //leap how far again?
+	var/throw_speed = SPEED_AVERAGE
+	var/windup = FALSE // Is there a do_after before we pounce?
+	var/windup_duration = 20 // How long to wind up, if applicable
```
dm doc 

so do
/// How far does this leap go?

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -607,3 +722,6 @@ CULT
 
 	var/mob/living/carbon/human/human_user = owner
 	SEND_SIGNAL(human_user, COMSIG_MOB_MG_EXIT)
+
+
+
```
```suggestion
```

harryob commented on code/modules/mob/living/carbon/human/human_abilities.dm:
```diff
@@ -479,6 +491,114 @@ CULT
+		return
+	var/mob/living/carbon/human/victim = M
+	var/datum/disease/black_goo/located_disease = locate() in victim.viruses
+	if(located_disease)
+		to_chat(zombie, SPAN_XENOWARNING("The target is already infected, why would you bite it again?"))
+	else
+		enter_cooldown(10 SECONDS)
+		to_chat(zombie, SPAN_XENOWARNING("You reach down to [M] neck, preparing to bite it!"))
+		to_chat(M, SPAN_DANGER("[zombie.name] reaches your neck and starts to open [zombie.p_their()] jaw!"))
+		if(!do_after(zombie, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, M, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
+			to_chat(zombie, SPAN_XENOWARNING("You were interupted!"))
+			return
+		to_chat(zombie, SPAN_XENOWARNING("You bite [M]'s neck, leaving a bloody mark!"))
+		to_chat(M, SPAN_DANGER("[zombie] bites you right in the neck!"))
+		playsound(get_turf(zombie), 'sound/hallucinations/wail.ogg', 25, TRUE)
+		victim.emote("scream")
+		playsound(victim, "bone_break" , 45, TRUE)
+		victim.AddDisease(new /datum/disease/black_goo)
+		zombie.flick_attack_overlay(victim, "bite")
+		zombie.animation_attack_on(victim)
+		victim.apply_damage(20, BRUTE, "head")
```
```suggestion
	if(located_disease)
		to_chat(zombie, SPAN_XENOWARNING("The target is already infected, why would you bite it again?"))
		return

	enter_cooldown(10 SECONDS)
	to_chat(zombie, SPAN_XENOWARNING("You reach down to [M] neck, preparing to bite it!"))
	to_chat(M, SPAN_DANGER("[zombie.name] reaches your neck and starts to open [zombie.p_their()] jaw!"))
	
	if(!do_after(zombie, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, M, INTERRUPT_MOVED, BUSY_ICON_HOSTILE))
		to_chat(zombie, SPAN_XENOWARNING("You were interupted!"))
		return
	
	to_chat(zombie, SPAN_XENOWARNING("You bite [M]'s neck, leaving a bloody mark!"))
	to_chat(M, SPAN_DANGER("[zombie] bites you right in the neck!"))
	playsound(get_turf(zombie), 'sound/hallucinations/wail.ogg', 25, TRUE)
	victim.emote("scream")
	playsound(victim, "bone_break" , 45, TRUE)
	victim.AddDisease(new /datum/disease/black_goo)
	zombie.flick_attack_overlay(victim, "bite")
	zombie.animation_attack_on(victim)
	victim.apply_damage(20, BRUTE, "head")
```

harryob commented on code/modules/reagents/chemistry_properties/prop_positive.dm:
```diff
@@ -900,6 +900,34 @@
+
+/datum/chem_property/positive/antibiotic/process(mob/living/M, potency = 1, delta_time)
+	M.apply_damage(0.5 * potency * delta_time, TOX)
+	var/datum/disease/black_goo/located_disease = locate() in M.viruses
+	if(located_disease)
+		var/datum/disease/black_goo/virus = new
+		var/mob/living/carbon/human/infected = M
+		if(virus.stage == 1)
+			if(prob(3 * potency))
+				for(var/datum/disease/goo in infected.viruses)
+					goo.cure()
+			else
+				if(prob(15 * potency))
+					to_chat(M, SPAN_WARNING("Your insides feel insanely hot!"))
```
```suggestion
	var/datum/disease/black_goo/located_disease = locate() in M.viruses
	if(!located_disease)
		return
		
	if(located_disease.stage != 1)
		return
	
	if(prob(3 * potency))
		for(var/datum/disease/black_goo/goo in M.viruses)
			goo.cure()
		return

	if(prob(15 * potency))
		to_chat(M, SPAN_WARNING("Your insides feel insanely hot!"))
```

harryob commented on code/modules/reagents/chemistry_properties/prop_positive.dm:
```diff
@@ -900,6 +900,34 @@
+	M.apply_damage(4 * potency * delta_time, TOX)
+
+/datum/chem_property/positive/antibiotic/process_critical(mob/living/M, potency, delta_time)
+	M.apply_damage(8 * potency * delta_time, TOX)
```
```suggestion
	M.apply_damage(8 * potency * delta_time, TOX)

```

kiVts commented:
commits above fix dead zombies making things they shouldnt. 
still on the hunt for the one hour revive timer

kiVts commented:
this fixes one hour revive timer

kiVts commented:
![image](https://user-images.githubusercontent.com/48099872/230098384-e15ad5ba-ef98-4898-b1b5-3deda5ce419d.png)

I'm just going to kill this sooner than later by myself because that's what they meant. For whatever reason PR that makes zombies better after someone attempts to remove them is not a good enough reason. (^morrow said the text in screenshot btw) + Im tired of this dumb shit. 

Sooo.. Enjoy your sweet zombies?

SarahJohnson489 commented:
We just got the 1 hr revive bug in the last zombies round. If you could do an atomic PR for that it'd be great.

kiVts commented:
It involves removing unique and override flags from zombie revive timer, No I'm not doing it.
